### PR TITLE
Validate scriban on new notifications

### LINKED ISSUE #2837: Validate notification config templates at creation time
With the move to scriban templates, we have the opportunity to validate newly created notification configs before we store them in the database.

This will make sure we have only valid configs.

AB#43373524

### PR DESCRIPTION
**Pull request by tevoinea**
## Summary of the Pull Request

_What is this about?_

Closes #2837 

We should validate notification configs at creation time that they can be safely rendered by scriban.

* Move some scriban conversion utility functions to JinjaTemplateAdapter.cs
* Validate templates are compatible with scriban at notification creation time by:
    * Seeing if we need to apply any migrations to it
    * It can be rendered by scriban without throwing
* Update integration tests to support using feature flags

### CODE REVIEW

File: src/ApiService/ApiService/onefuzzlib/notifications/JinjaTemplateAdapter.cs [hunk 1]
```diff
@@ -11,4 +11,351 @@ public class JinjaTemplateAdapter {
...
+
+        var renderedTemaplate = await renderer.Render(template, new Uri(instanceUrl), strictRendering: true);
+
+        return new TemplateValidationResponse(
+            renderedTemaplate,
+            templateRenderContext
+        );
+    }
+
+    private static async Async.Task<(NotificationsBase.Renderer, TemplateRenderContext)> GenerateTemplateRenderContext(IOnefuzzContext context, ILogTracer log, TemplateRenderContext? templateRenderContext) {
```
Comment (tevoinea) [inline src/ApiService/ApiService/onefuzzlib/notifications/JinjaTemplateAdapter.cs:47]: This function was copied directly from `src/ApiService/ApiService/Functions/ValidateScriban.cs`

File: src/ApiService/ApiService/onefuzzlib/notifications/JinjaTemplateAdapter.cs [hunk 2]
```diff
@@ -11,4 +11,351 @@ public class JinjaTemplateAdapter {
...
+            targetUrl,
+            reportContainer,
+            reportFileName,
+            reproCmd
+        );
+
+        return (renderer, templateRenderContext);
+    }
+
+    public async static Async.Task<(bool didModify, AdoTemplate template)> ConvertToScriban(AdoTemplate template, bool attemptRender = false, IOnefuzzContext? context = null, ILogTracer? log = null) {
```
Comment (tevoinea) [inline src/ApiService/ApiService/onefuzzlib/notifications/JinjaTemplateAdapter.cs:162]: This was copied from `src/ApiService/ApiService/Functions/Migrations/JinjaToScriban.cs`, but I added the option render related args.

File: src/ApiService/ApiService/onefuzzlib/notifications/JinjaTemplateAdapter.cs [hunk 3]
```diff
@@ -11,4 +11,351 @@ public class JinjaTemplateAdapter {
...
+                };
+            } else if (attemptRender) {
+                await ValidateScribanTemplate(context!, log!, null, template.OnDuplicate.Comment).IgnoreResult();
+            }
+        }
+
+        return (didModify, template);
+    }
+
+    public async static Async.Task<(bool didModify, GithubIssuesTemplate template)> ConvertToScriban(GithubIssuesTemplate template, bool attemptRender = false, IOnefuzzContext? context = null, ILogTracer? log = null) {
```
Comment (tevoinea) [inline src/ApiService/ApiService/onefuzzlib/notifications/JinjaTemplateAdapter.cs:233]: This was copied from `src/ApiService/ApiService/Functions/Migrations/JinjaToScriban.cs`, but I added the option render related args.

File: src/ApiService/ApiService/Functions/Migrations/JinjaToScriban.cs [hunk 4]
```diff
@@ -39,11 +39,11 @@ public class JinjaToScriban {
         _log.Info($"Finding notifications to migrate");
 
         var notifications = _context.NotificationOperations.SearchAll()
-            .Select(notification => {
+            .SelectAwait(async notification => {
```
Comment (chkeita) [inline src/ApiService/ApiService/Functions/Migrations/JinjaToScriban.cs:42]: non blocking comment 
you have the opportunity here to convert those templates in parallel. depending on the size of the dataset that could speed things up.

**Review by chkeita — APPROVED** [review]

**Pull request merged**